tools/codegen/model.py

```diff
@@ -1496,3 +1509,38 @@ def parse_returns(return_decl: str) -> Tuple[Return, ...]:
     if return_decl[0] == '(' and return_decl[-1] == ')':
         return_decl = return_decl[1:-1]
     return tuple(Return.parse(arg) for arg in return_decl.split(', '))
+
+
+# A Precompute instance consists of a map from kernel argument name
+# to the list of Argument instances that should replace that
+# kernel argument in the impl function.
+@dataclass(frozen=True)
+class Precompute:
+    # A map from kernel argument name -> a list of precomputed
+    # elements that replaces/supersedes it.
+    replace: Dict[str, List[Argument]]
+
+    @staticmethod
+    def parse(src: List[str]) -> 'Precompute':
```

Comment:
This causes a mypy error because the annotated type of the object that gets passed to this is `object`. Should I use `Any`/`object` instead and add `isinstance` checks to make sure it is a `list` of `str`?

Reply:
yep. this is because the yaml is completely untyped and you gotta test if it's the right type